src/corell/src/lib.rs

```diff
@@ -123,6 +135,8 @@ pub struct Device<R: Resources, F: Factory<R>, Q: CommandQueue> {
     pub transfer_queues: Vec<TransferQueue<Q>>,
     pub heap_types: Vec<HeapType>,
     pub memory_heaps: Vec<u64>,
+    /// Device capabilities
+    pub caps: Capabilities,
```

Comment:
That was my original design, but unfortunately D3D12 only queries those via `D3D12Device`, which we only get in `open()`, so I had to move stuff under `Device`. Anyhow, I'll be happy to revise this later.